client/alloc_runner.go

```diff
@@ -433,15 +456,22 @@ OUTER:
 			for _, tr := range runners {
 				tr.Update(update)
 			}
+		case <-watchdog.C:
+			if event, desc := r.checkResources(); event != nil {
+				r.setStatus(structs.AllocClientStatusFailed, desc)
+				taskDestroyEvent = event
+				break OUTER
+			}
 		case <-r.destroyCh:
+			taskDestroyEvent = structs.NewTaskEvent(structs.TaskKilled)
```

Comment:
I would set `taskDestroyEvent = nil` here and at line 450. See comment on TaskRunner